nixos/modules/config/theming.nix

```diff
@@ -0,0 +1,320 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.theming;
+  gcfg = config.theming.gtk3;
+  qcfg = config.theming.qt5;
+  theme = config.theming.theme;
+
+  toGtk3Ini = generators.toINI {
+    mkKeyValue = key: value:
+      let
+        value' =
+          if isBool value then (if value then "true" else "false")
+          else toString value;
+      in
+        "${key}=${value'}";
+  };
+  toDconfIni = generators.toINI {
+    mkKeyValue = key: value:
+      let
+        value' =
+          if isBool value then (if value then "true" else "false")
+          else toString value;
+      in
+        "${key}='${value'}'";
+  };
+
+  fontType = types.submodule {
+    options = {
+      package = mkOption {
+        internal = true;
+        type = types.nullOr types.package;
+        default = null;
+      };
+      name = mkOption {
+        internal = true;
+        type = types.str;
+      };
+    };
+  };
+  themeType = types.submodule {
+    options = {
+      package = mkOption {
+        internal = true;
+        type = types.nullOr types.package;
+        default = null;
+      };
+      name = mkOption {
+        internal = true;
+        type = types.str;
+      };
+    };
+  };
+
+  optionalPackage = opt:
+    optional (opt != null && opt.package != null) opt.package;
+in
+{
+  options = {
+    theming = {
+      gtk3 = {
+        enable = mkEnableOption "Gtk3 theming configuration";
+
+        font = mkOption {
+          type = types.nullOr fontType;
+          default = null;
+          description = ''
+            The font to use in GTK+ 3 applications.
+          '';
+        };
+
+        iconTheme = mkOption {
+          type = types.nullOr themeType;
+          default = null;
+          description = "The icon theme to use.";
+        };
+
+        theme = mkOption {
+          type = types.nullOr themeType;
+          default = null;
+          description = "The GTK+3 theme to use.";
+        };
+      };
+
+      qt5 = {
```

Comment:
There's a `qt5.nix`, perhaps the qt specific code should go there.

Reply:
@worldofpeace I plan to add support for gtk2 and qt4 (e.g. GIMP still uses gtk2 and we still have qt4 packages), it will be probably simpler to keep it in one place then